lib/kitchen/provisioner/ansible_push.rb

```diff
@@ -123,111 +251,41 @@ def exec_ansible_command(env, command, desc)
         @instance_connection_option = instance.transport.instance_variable_get(:@connection_options)
         debug("instance_connection_option=" + @instance_connection_option.to_s)
-        hostname =  if @instance_connection_option.nil?
-                       @machine_name
+        @instance_connection_option
+      end
+
+      def prepare_inventory
+        hostname =  if instance_connection_option().nil?
+                       machine_name
                     else
-                        @instance_connection_option[:hostname]
+                        instance_connection_option()[:hostname]
                     end
         debug("hostname=" + hostname)
-        write_instance_inventory(@machine_name, hostname, config[:mygroup], @instance_connection_option)
-      end
-
-      def compile_config()
-        debug("compile_config")
-        options = []
-        options << "--extra-vars='#{self.get_extra_vars_argument}'" if config[:extra_vars]
-        options << "--sudo" if config[:sudo]
-        options << "--sudo-user=#{config[:sudo_user]}" if config[:sudo_user]
-        options << "--user=#{config[:remote_user]}" if self.get_remote_user
-        options << "--private-key=#{config[:private_key]}" if config[:private_key]
-        options << "#{self.get_verbosity_argument}" if config[:verbose]
-        options << "--diff" if config[:diff]
-        options << "--ask-sudo-pass" if config[:ask_sudo_pass]
-        options << "--ask-vault-pass" if config[:ask_vault_pass]
-        options << "--vault-password-file=#{config[:vault_password_file]}" if config[:vault_password_file]
-        options << "--tags=%s" % self.as_list_argument(config[:tags]) if config[:tags]
-        options << "--skip-tags=%s" % self.as_list_argument(config[:skip_tags]) if config[:skip_tags]
-        options << "--start-at-task=#{config[:start_at_task]}" if config[:start_at_task]
-        options << "--inventory-file=`which kitchen-ansible-inventory`" if config[:generate_inv]
-        ##options << "--inventory-file=#{ssh_inv}," if ssh_inv
-        
-        # By default we limit by the current machine,
-        if config[:limit]
-          options << "--limit=#{as_list_argument(config[:limit])}"
-        else
-          options << "--limit=#{@machine_name}"
-        end
-      
-        #Add raw argument as final thing
-        options << config[:raw_arguments] if config[:raw_arguments]
-
-        @command = (%w(ansible-playbook) << options << config[:playbook]).flatten.join(" ")
-        debug("Ansible push command= %s" % @command)
-        @command_env = {
-          "PYTHONUNBUFFERED" => "1", # Ensure Ansible output isn't buffered
-          "ANSIBLE_FORCE_COLOR" => "true",
-          "ANSIBLE_HOST_KEY_CHECKING" => "#{config[:host_key_checking]}",
-        }
-        @command_env["ANSIBLE_CONFIG"]=config[:ansible_config] if config[:ansible_config]
-        info("Ansible push compile conig done")
-      end
-
-      def validate_config()
-        if !config[:playbook]
```

Comment:
@mcamou  all this code already existed, I just moved it around and extracted functions, it doesn't seem like making further changes of coding style would be relevant to this specific pull request?